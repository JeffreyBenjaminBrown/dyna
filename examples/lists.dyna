f([1,2]).

a := [1,2].

goal(X) := f([1|X]).

% structured term, sould be true
b := &f("a") in [1,2,&f("a"),3].

% should be true
c := 2 in [1,2,3].

% should be false
d := 4 in [1,2,3].

% simpler iteration of a complex list
things set= X for X in [1,[2,2],[3,4]].

% unpack structure requiring a check
d(&X) := true for [X,X] in [1,[2,2],[3,4],[4,4]].

% these two examples should an interesting behavior
foo(A) := true for &f(A) in [1,2,&f("a"),3].

% this one checks if the value of f(A) is in the list, (note: 1 == True, in python).
goo(A) := true for f(A) in [1,2,&f("a"),3].

thingsbag bag= "three".
thingsbag bag= 1.
thingsbag bag= 1.
thingsbag bag= 2.

thingset set= "three".
thingset set= 1.
thingset set= 1.
thingset set= 2.