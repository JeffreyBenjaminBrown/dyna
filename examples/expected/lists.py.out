
Solution
========
a => [1, 2].

f/1
===
f([1, 2]) => true.

goal/1
======
goal([2]) => true.

